### PR TITLE
New: Link indexer to specific download client

### PR DESCRIPTION
**Pull request by cqxmzz**
#### Database Migration
YES - 164 (DownloadClientId on Indexers)

#### Description
Allow user to specify a particular client for an indexer
Basically copied from this radarr change: 
https://github.com/Radarr/Radarr/pull/6858

#### Todos
- [x] Tests
- [ ] Wiki Updates

#### Issues Fixed or Closed by this PR

Closes [#1215](https://github.com/Sonarr/Sonarr/issues/1215)

### CODE REVIEW

**Comment by bakerboy448** [comment]
Tests failed

https://build.sonarr.tv/viewLog.html?buildId=172268&buildTypeId=Sonarr_V3PullRequests_5TriggerBuild

(Hit login as guest if you hit the auth wall)

also should the commit be prefixed with "New:"

it's also typically best practice to cherry-pick it; or at least add the original committer as a co author in the commit ;)

**Comment by cqxmzz** [comment]
> Tests failed
> 
> https://build.sonarr.tv/viewLog.html?buildId=172268&buildTypeId=Sonarr_V3PullRequests_5TriggerBuild
> 
> (Hit login as guest if you hit the auth wall)
> 
> also should the commit be prefixed with "New:"
> 
> it's also typically best practice to cherry-pick it; or at least add the original committer as a co author in the commit ;)

Thanks! It's my first pull request ever.
Let me try to fix all of those.

**Comment by markus101** [comment]
The original commit author should be maintained, and subsequent changes needed to move from Radarr to Sonarr should be squashed into that commit.

File: src/NzbDrone.Core/Datastore/Migration/163_download_client_per_indexer.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(163)]
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/163_download_client_per_indexer.cs:6]: This is already used in another PR:

https://github.com/Sonarr/Sonarr/pull/4815

**Comment by cqxmzz** [comment]
> The original commit author should be maintained, and subsequent changes needed to move from Radarr to Sonarr should be squashed into that commit.

I added the co-author, anything else I should do for this?

**Comment by bakerboy448** [comment]
Git nit in the OP, but for it to be properly linked the GitHub issue it's closing should be prefixed with Closes or Fixes

E.g. Closes #1215

**Comment by markus101** [comment]
Thanks for pulling this over!

**Pull request merged**